autoload/coc/powershell.vim

```diff
@@ -1,15 +1,28 @@
 let s:root           = expand('<sfile>:h:h:h')
 let s:is_win         = has('win32') || has('win64')
-let s:is_mac = !s:is_win && !has('win32unix')
+let s:is_mac         = !s:is_win && !has('win32unix')
       \ && (has('mac') || has('macunix') || has('gui_macvim') ||
       \   (!isdirectory('/proc') && executable('sw_vers')))
 let s:is_vim         = !has('nvim')
-let s:install_script = s:root.'/'.(s:is_win ? 'install.cmd' : 'install.ps1')
+if(!exists("g:coc_powershell_prerelease"))
```

Comment:
Instead of this, we could have Install() take in a parameter called `AllowPrerelease` or something and then use that to determine if the install script gets called with `AllowPrerelease`?

Reply:
If `coc#powershell#install()` took in a Dictionary, then we could have:

```
{
    AllowPrerelease: bool,
    PowerShellExecutable: string
}
```

But I won't block on this

Reply:
Ooooh I like that better. Then I don't need to add extra settings to my vimrc, I can just change the post-install action 😄

Reply:
Exactly! It's like an optional configuration object 😊

Reply:
The only "issue" with this method is that you have to specify something to the function or it complains. It can be an empty dictionary (as the README has been updated to include). I'll do some more research in the morning on if we can make it an optional variable or something.

Reply:
You want to use [method/function overloading](https://pythonspot.com/method-overloading/) where you have a
```
coc#powershell#install()
```
And 
```
coc#powershell#install(dict)
```
Where the first calls the second with an empty dictionary.

Or see if vim supports default parameters.

Reply:
Would love to get @yatli's thoughts.

Reply:
Hopefully there's some other way to do function overloading in vimscript...

![image](https://user-images.githubusercontent.com/30301021/56439010-94f34500-6299-11e9-80ea-f53bb09d97a7.png)

Reply:
I think variadic function `function! Mine(...)` is the way to go.

you can then check the number of args with `a:0`

Reply:
Yeah, I switched it to using variadic function, but only check if `a:1` contains the dictionary for flags and PowerShell executable.